pkg/kv/kvserver/replica_application_state_machine.go

```diff
@@ -302,9 +259,9 @@ func (b *replicaAppBatch) Stage(
 	// way, it would become less of a one-off.
 	if splitMergeUnlock, err := b.r.maybeAcquireSplitMergeLock(ctx, cmd.Cmd); err != nil {
 		if cmd.Cmd.ReplicatedEvalResult.Split != nil {
-			err = wrapWithNonDeterministicFailure(err, "unable to acquire split lock")
+			err = kvserverbase.NonDeterministicFailureErrorWrapf(err, "unable to acquire split lock")
```

Comment:
nit: How about `NonDeterministicFailureWrapf`

Reply:
I had that at first but then I felt dirty because the type is called `NonDeterministicFailureWrapf`. Maybe should just call this thing `NonDeterministicError`? `FailureError` is awfully redundant.

Reply:
Agree that `Error` is a gooder name than `Failure`, it's shorter and alludes to the fact that it implements Go `error`. Also sometimes I see `ErrBlah`, but more for exported variables than types.